src/gopher.cc

```diff
@@ -102,6 +102,9 @@ class GopherStateData
 
     ~GopherStateData();
 
+    /// URL for icon to display, given the item-type code
```

Comment:
We should document the lifetime of the returned-by-pointer object. We should not document how the URL will be _used_ -- that is up to the caller:

```suggestion
    /// MIME type icon URL for the given menu item type (or nil). The returned
    /// c-string is invalidated by the next call to this or similar functions.
```

or, if you prefer a more specific/accurate but less convenient variant (that requires the reader to start chasing a long list of poorly documented functions that all share the same lifetime validity flaws):

```suggestion
    /// MIME type icon URL for the given menu item type (or nil). See
    /// mimeGetIconURL() for the lifetime of the returned value.
```

I do not insist on any specific text here, of course.

Reply:
Added the lifetime mention in [89be89e](https://github.com/squid-cache/squid/pull/1026/commits/89be89e11f0dc0fbf6f93776fd2c2c17b40e720c).

I am not mentioning "mime" or "menu" as those concepts are irrelevant to this method API or caller(s). This is simply an icon URL (not necessarily mime related), and the input is a Gopher "item-type" code (not restricted to menus).